charts/spire/charts/spire-server/templates/issuer.yaml

```
@@ -0,0 +1,47 @@
+{{- with .Values.upstreamAuthority.certManager }}
+{{ if and .enabled .createCA }}
+{{/*
+Configuring CA Issuer: https://cert-manager.io/docs/configuration/ca/
+*/}}
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: {{ include "spire-server.fullname" $ }}-signed
+  namespace: {{ include "spire-server.namespace" $ }}
+  labels:
+    {{- include "spire-server.labels" $ | nindent 4}}
+spec:
+{{- if .signedSpec }}
+  {{- toYaml .signedSpec | nindent 4 }}
+{{- else }}
+  selfSigned: {}
+{{- end }}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: {{ include "spire-server.fullname" $ }}-ca
+  namespace: {{ include "spire-server.namespace" $ }}
+spec:
+  isCA: true
+  commonName: {{ $.Values.ca_subject.common_name }}
+  secretName: {{ include "spire-server.fullname" $ }}-ca-keys
+  privateKey:
+    algorithm: ECDSA
+    size: 256
+  issuerRef:
+    name: {{ include "spire-server.fullname" $ }}-signed
+    kind: Issuer
+    group: cert-manager.io
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: {{ include "spire-server.fullname" $ }}-ca
```

Aligning this with https://github.com/drewwells/helm-charts/blob/certmanager/charts/spire/charts/spire-server/templates/configmap.yaml#L81 could be problematic as chart nesting or name overrides could affect this....

It may be easier to just keep
issuer_name (https://github.com/drewwells/helm-charts/blob/certmanager/charts/spire/charts/spire-server/templates/configmap.yaml#L81) set as it was, and
here
```suggestion
  name: {{ upstreamAuthority.certManager.issuer_name }}
```

Then the user is free to override it to whatever they want, in the rare case where you have two upstreamCA.certmanager things in the same namespace wanting to use the same ca. (not sure that even works anyway. The ca certificate minted by spire may have name conflicts by the plugin.

---

This has come up already once before, I've very adamant that naming this with`.Chart.fullname` is important. 5+ years of helm usage has taught me that lesson.

If the user wants the Issuer to be a different name, they can do so by changing the release name `helm install mySpecialName`. I doubt anyone has or will request that feature. Even if they do, this entire object is in a guard statement so user can create an entirely different issuer.